addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -146,13 +149,16 @@ options.registry.SocialMedia = options.Class.extend({
                 const href = anchorEl.getAttribute('href');
                 if (href !== entry.display_name) {
                     if (this._isValidURL(entry.display_name)) {
-                        // Propose an icon only for valid URLs (no mailto).
+                        // Propose an icon only for valid URLs (no mailto) and
+                        // only if there is already an icon (so not an image).
                         const socialMedia = this._findRelevantSocialMedia(entry.display_name);
                         if (socialMedia) {
                             const iEl = anchorEl.querySelector('i');
                             this._removeSocialMediaClasses(anchorEl);
-                            anchorEl.classList.add(`s_social_media_${socialMedia}`);
-                            iEl.classList.add(`fa-${socialMedia}`);
+                            if (iEl) {
+                                anchorEl.classList.add(`s_social_media_${socialMedia}`);
```

Comment:
I am curious:
- Why only adding this class if there is an icon?
- Already like that before butu: why calling `_removeSocialMediaClasses` only if we found a social media?

Reply:
> Why only adding this class if there is an icon

From what I saw, the `s_social_media_...` class only purpose is to set the icon color to the one matching the social media used (like the facebook blue, or google red) so it will not break anything on elements other than icons but it will be useless (so better not use it) 
![image](https://github.com/odoo/odoo/assets/102796037/45dc0070-26c0-4287-a943-c36ab9d85711)

> why calling _removeSocialMediaClasses

This function is used to clean the previous social media classes before setting the new ones to avoid "clashes" between them when setting the new matching classes (so remove anything "facebook-related" before adding the "google-specific") given that the `fa` classes manage the icon used in the "icon" (and `s_social_media_...` the color as said previously)

Note that this function is also called when adding a new icon (above, before the first diff), to clean it since it is a copy of the first one  
But also note that replacing the icon (so with the media dialog) will not remove the `s_social_media_` class from the anchor element, so I think we remove the classes only if really necessary :slightly_smiling_face:

Reply:
> so it will not break anything on elements other than icons but it will be useless (so better not use it)

I have the opposite opinion: the fact it is useless in some configuration is not important. What is better IMO is that if the option detects it is a facebook URL, the related class is added, always, whatever its effect is. A custo could take profit of that (and actually, as we are in stable, best change the behavior as less as possible) :+1: 

> to clean the previous social media classes before setting the new ones

Yes, but since you now were seting the new one only `if (iEl)`, it seems weird: either cleaning should be in that `if` too or outside of the main one I think :wink: But as I said above, I think we should rather add the  `s_social_media_XXX` class as soon as we find a social media :+1: